card/lib/card/model/save_helper.rb

```diff
@@ -115,7 +115,7 @@ def add_name name, content_or_args
 
       def resolve_name_conflict args
         rename = args.delete :rename_if_conflict
-        return unless rename
+        return unless args[:name] && rename
```

Comment:
not a problem that you created here, but I think this should have one arg ("name") rather than a hash.